python/src/aac/lang/lsp/providers/goto_definition_provider.py

```diff
@@ -0,0 +1,50 @@
+"""Module for the Goto Definition Provider which handles all goto definition requests."""
+
+from pygls.server import LanguageServer
+from pygls.lsp.types.basic_structures import Location, Position, Range
+from pygls.lsp.types.language_features.definition import DefinitionParams
+from pygls.workspace import Document
+
+from aac.lang.lsp.providers.lsp_provider import LspProvider
+
+
+class GotoDefinitionProvider(LspProvider):
+    """Resolve the location where a specified name is defined."""
+
+    def handle_request(self, ls: LanguageServer, params: DefinitionParams) -> list[Location]:
+        """Return the location at which the specified item is found."""
+        return self.get_definition_location(ls.workspace.documents, params.text_document.uri, params.position)
+
+    def get_definition_location(self, documents: dict[str, Document], current_uri: str, position: Position) -> list[Location]:
+        """Return the location where the AaC definition is defined."""
+        name = documents.get(current_uri).word_at_position(position)
+        if not name:
+            return []
+
+        locations = []
+        for doc in documents.values():
+            ranges = self.get_ranges_containing_name(doc.source, name)
```

Comment:
I'm catching some errant definition goto suggestions, we should make this more explicit to avoid bad suggestions. For example, when I goto definition for `enum`, I get the schema definition for `enum`, the root key entry for `enum`, and the schema definition for `enumExt`. 

The cleaner behavior should be:
A. goto definition on a reference to a type goes to the type's schema definition.
B. goto definition on a root key goes to the root-key declaration.

Reply:
Okay, I fixed this but I realized that I'm not handling enumerated value definitions correctly. I'll work on that tomorrow.